Sources/Plasma/Apps/plClient/Mac-Cocoa/plMacDisplayHelper.h

```diff
@@ -0,0 +1,76 @@
+/*==LICENSE==*
+
+CyanWorlds.com Engine - MMOG client, server and tools
+Copyright (C) 2011  Cyan Worlds, Inc.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+Additional permissions under GNU GPL version 3 section 7
+
+If you modify this Program, or any covered work, by linking or
+combining it with any of RAD Game Tools Bink SDK, Autodesk 3ds Max SDK,
+NVIDIA PhysX SDK, Microsoft DirectX SDK, OpenSSL library, Independent
+JPEG Group JPEG library, Microsoft Windows Media SDK, or Apple QuickTime SDK
+(or a modified version of those libraries),
+containing parts covered by the terms of the Bink SDK EULA, 3ds Max EULA,
+PhysX SDK EULA, DirectX SDK EULA, OpenSSL and SSLeay licenses, IJG
+JPEG Library README, Windows Media SDK EULA, or QuickTime SDK EULA, the
+licensors of this Program grant you additional
+permission to convey the resulting work. Corresponding Source for a
+non-source form of such a combination shall include the source code for
+the parts of OpenSSL and IJG JPEG Library used as well as that of the covered
+work.
+
+You can contact Cyan Worlds, Inc. by email legal@cyan.com
+ or by snail mail at:
+      Cyan Worlds, Inc.
+      14617 N Newport Hwy
+      Mead, WA   99021
+
+*==LICENSE==*/
+
+#ifndef plMacDisplayHelper_hpp
+#define plMacDisplayHelper_hpp
+
+#include <stdio.h>
+// Currently requires Metal to query attached GPU capabilities
+// Capability check will also work for GL - but will need something
+// different for older GPUs.
+#include <AppKit/AppKit.h>
+#include <Metal/Metal.hpp>
+#include <QuartzCore/QuartzCore.h>
+
+#include "plPipeline/hsG3DDeviceSelector.h"
+
+class plMacDisplayHelper: public plDisplayHelper
+{
+public:
+    CGDirectDisplayID CurrentDisplay() { return fCurrentDisplay; }
+    // we need NSScreen to query for non rectangular screen geometry
+    void SetCurrentScreen(hsDisplayHndl screen);
+    void SetCurrentScreen(NSScreen *screen);
+
+    plDisplayMode DesktopDisplayMode() override { return fDesktopDisplayMode; };
+    std::vector<plDisplayMode> GetSupportedDisplayModes(hsDisplayHndl display, int ColorDepth = 32) override;
+    
+    plMacDisplayHelper();
+    ~plMacDisplayHelper();
+private:
+    CGDirectDisplayID fCurrentDisplay = -1;
+    MTL::Device* fRenderDevice = nullptr;
```

Comment:
This doesn't ever appear to get initialized or assigned a value, and my preference would be to avoid introducing Metal-isms into shared code as much as possible

Reply:
plMacDisplayHelper used to decide the default resolution - but I farmed that back out to the Metal enumerator. I've removed the remnants of this.

Do you have any opinions here? That would leave the GL renderer also responsible for deciding the default resolution. But that might be where that responsibility belongs.